sonar-kotlin-plugin/src/main/java/org/sonarsource/kotlin/api/regex/RegexContext.kt

```diff
@@ -27,24 +27,36 @@ import org.sonarsource.analyzer.commons.regex.ast.FlagSet
 import org.sonarsource.analyzer.commons.regex.ast.RegexSyntaxElement
 import org.sonarsource.kotlin.plugin.KotlinFileContext
 
+class RegexCache(
+    private val globalCache: MutableMap<Pair<List<KtStringTemplateEntry>, Int>, RegexParseResult> = mutableMapOf()
+) {
+    fun add(
```

Comment:
It seems more descriptive to me to name this `computeIfAbsent` too - or maybe at least `addIfAbsent` or even `parseAndCacheIfAbsent` or something along those lines. That way the caller gets a hint that more is happening under the hood than just an `add` operation.